src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourcePane.java

```diff
@@ -150,6 +150,11 @@ public void closeTab(int index, boolean interactive, Command onClosed)
          tabPanel_.closeTab(index, onClosed);
    }
    
+   public void closeAllTabs()
+   {
+      tabPanel_.clear();
```

Comment:
This is another instance of bypassing higher-level APIs (and their semantics and side effects). If you look at the DocTabLayoutPanel.remove method (which is ultimately called by closeTab) you can see the side effects there that we want to preserve (for now that that's firing TabClosed events but other side effects could be added). What you really want here is the same code that's in closeAllSourceDocs:

```java
cpsExecuteForEachEditor(editors_, new CPSEditingTargetCommand()
{
   @Override
   public void execute(EditingTarget target, Command continuation)
   {
      view_.closeTab(target.asWidget(), false, continuation);
   }
});
```